cmake/JuliaDeps.cmake

```diff
@@ -0,0 +1,16 @@
+jl_str_option(LLVM_VER "" 3.3)
+jl_str_option(LLVM_LIB_SUFFIX "" "")
+jl_str_option(PCRE_VER "" 10.10)
+jl_str_option(DSFMT_VER "" 2.2.3)
+jl_str_option(LAPACK_VER "" 3.5.0)
+jl_str_option(ARPACK_VER "" 3.2.0)
+jl_str_option(FFTW_VER "" 3.3.4)
+jl_str_option(SUITESPARSE_VER "" 4.4.2)
+jl_str_option(UNWIND_VER "" 1.1)
+jl_str_option(OSXUNWIND_VER "" 0.0.1)
+jl_str_option(GMP_VER "" 6.0.0)
+jl_str_option(MPFR_VER "" 3.1.2)
+jl_str_option(PATCHELF_VER "" 0.8)
+jl_str_option(GIT_VER "" 1.8.5.6)
+jl_str_option(VIRTUALENV_VER "" 1.11.6)
+jl_str_option(RMATH_JULIA_VER "" 0.1)
```

Comment:
One issue I've realized for making `Makefile` and `cmake` coexisting is that we either can't use `configure_file` in cmake or have to duplicate the logic. (Although it is possible to do...)

For this particular issue (dependency version) I would probably prefer having them defined in a text file and have a cmake target to generate the necessary files for Makefile if necessary. I don't expect these versions to change very often so having the person updating it to run a cmake command after the update when we have both build system shouldn't be too hard and is slightly less hacky on both side.